templates/default.tpl

```diff
@@ -1,362 +1,138 @@
-});
-$(document).ready(function () {
-    $(".time").text(function(_, text) {
-        return $.renderTime(text);
-      });
-    $(".citime").text(function(_, text) {
-        return $.renderTime(text);
-      });
-    $(".percent").text(function(_, text) {
-        return (text*100).toFixed(1);
-      });
-  });
-</script>
+<head>
+  <meta charset="utf-8"/>
+  <title>criterion report</title>
+  <script>
+    {{{js-chart}}}
+    {{{js-criterion}}}
+  </script>
+  <style>
+    {{{criterion-css}}}
+  </style>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+</head>
+<body>
+  <div class="content">
+    <h1 class='title'>criterion performance measurements</h1>
+    <p class="no-print"><a href="#grokularation">want to understand this report?</a></p>
+    <h1 id="overview"><a href="#overview">overview</a></h1>
+    <div class="no-print">
+      <select id="sort-overview" class="select">
+        <option value="report-index">index</option>
+        <option value="lex">lexical</option>
+        <option value="colex">colexical</option>
+        <option value="duration">time ascending</option>
+        <option value="rev-duration">time descending</option>
+      </select>
+      <span class="overview-info">
+        <a href="#controls-explanation" class="info" title="click bar/label to zoom; x-axis to toggle logarithmic scale; background to reset">&#9432;</a>
+        <a id="legend-toggle" class="chevron button"></a>
+      </span>
+    </div>
+    <aside id="overview-chart"></aside>
+    <main id="reports"></main>
 
-   </div>
+    <div id="report-data" data-report-json='{{{json}}}'></div>
   </div>
-  <div id="footer">
-    <div class="body">
-     <div class="footfirst">
-      <h2>colophon</h2>
-      <p>This report was created using the
-	<a href="http://hackage.haskell.org/package/criterion">criterion</a>
-	benchmark execution and performance analysis tool.</p>
-      <p>Criterion is developed and maintained
-      by <a href="http://www.serpentine.com/blog/">Bryan O'Sullivan</a>.</p>
-     </div>
+
+  <aside id="controls-explanation" class="explanation no-print">
+    <h1><a href="#controls-explanation">controls</a></h1>
+
+    <p>
+      The overview chart can be controlled by clicking the following elements:
+      <ul>
+        <li><em>a bar or its label</em> zooms the x-axis to that bar</li>
+        <li><em>the background</em> resets zoom to the entire chart</li>
+        <li><em>the x-axis</em> toggles between linear and logarithmic scale</li>
+        <li><em>the chevron</em> in the top-right toggles the the legend</li>
+        <li><em>a group name in the legend</em> shows/hides that group</li>
+      </ul>
+    </p>
+
+    <p>
+      The overview chart supports the following sort orders:
+      <ul>
+        <li><em>index</em> order is the order as the benchmarks are defined in criterion</li>
+        <li><em>lexical</em> order sorts groups left-to-right, alphabetically</li>
```

Comment:
Shouldn't it be "top-to-bottom", not "left-to-right", since the bars are arranged vertically?

Reply:
The left-to-right refers to that the groups are sorted with the leftmost being the most significant. Can you think of a better way to word it?

Reply:
I see what you mean, although I still think the status quo is a bit confusing. Ascending/descending order is perhaps the least ambiguous way of referring to the property of being sorted a certain way, so I would phrase this as:

> _lexical_ order sorts groups in ascending alphabetical order